File: CS/PS/Site/programmers/python/lv2/타겟넘버.py

def solution(numbers, target):
    answer = 0

    def dfs(sum, idx):
        result = 0
        if sum == target and idx == len(numbers):
            return 1
        if idx == len(numbers):
            return 0
        result += int(dfs(sum + numbers[idx], idx + 1))
        result += int(dfs(sum - numbers[idx], idx + 1))
        return result
    answer = dfs(0, 0)
    return answer

File: CS/PS/Site/programmers/lv2/타겟넘버.js
function solution(numbers, target) {
  var answer = 0;
  const dfs = (n, t, s, i) => {
    if (n.length <= i) {
      if (s === t) {
        answer++;
      }
      return;
    }
    dfs(n, t, s + n[i], i+1);
    dfs(n, t, s - n[i], i+1);
  };
  dfs(numbers, target, 0, 0);
  return answer;
}

console.log(solution([1, 1, 1, 1, 1]), 3);
